contracts/modules/assessment/Claims.sol

```diff
@@ -172,19 +172,27 @@ contract Claims is IClaims, RegistryAware {
 
       if (previousSubmission > 0) {
         (uint cooldownEnd, IAssessment.AssessmentStatus status) = _assessment().getAssessmentResult(previousSubmission);
-        
+
         require(
           status != IAssessment.AssessmentStatus.VOTING &&
-          status != IAssessment.AssessmentStatus.COOLDOWN, 
+          status != IAssessment.AssessmentStatus.COOLDOWN,
           ClaimIsBeingAssessed()
         );
 
-        require(
-          status == IAssessment.AssessmentStatus.DENIED ||
-          block.timestamp >= cooldownEnd + PAYOUT_REDEMPTION_PERIOD ||
-          _claims[claimId].payoutRedeemed,
-          PayoutCanStillBeRedeemed()
-        );
+        // Prevent re-submission for ACCEPTED claims
+        if (status == IAssessment.AssessmentStatus.ACCEPTED) {
+          // if payout already redeemed
+          require(
+            !_claims[previousSubmission].payoutRedeemed,
+            ClaimAlreadyPaidOut()
+          );
+
+          // if payout not yet redeemed and still within redemption period
+          require(
+            block.timestamp >= cooldownEnd + PAYOUT_REDEMPTION_PERIOD,
+            PayoutCanStillBeRedeemed()
+          );
+        }
```

Comment:
I refactored it to make it explicit what exactly are we guarding against.

I also added a new `ClaimAlreadyPaidOut` error if user is trying to double-dip